sqlglot/dialects/singlestore.py

```diff
@@ -1,5 +1,83 @@
+from sqlglot.dialects.dialect import build_formatted_time
 from sqlglot.dialects.mysql import MySQL
+import typing as t
+from sqlglot import exp, Dialect
+from sqlglot.helper import seq_get
 
 
 class SingleStore(MySQL):
     SUPPORTS_ORDER_BY_ALL = True
+
+    TIME_MAPPING: t.Dict[str, str] = {
+        "D": "%u",  # Day of week (1-7)
+        "DD": "%d",  # day of month (01-31)
+        "DY": "%a",  # abbreviated name of day
+        "HH": "%I",  # Hour of day (01-12)
+        "HH12": "%I",  # alias for HH
+        "HH24": "%H",  # Hour of day (00-23)
+        "MI": "%M",  # Minute (00-59)
+        "MM": "%m",  # Month (01-12; January = 01)
+        "MON": "%b",  # Abbreviated name of month
+        "MONTH": "%B",  # Name of month
+        "SS": "%S",  # Second (00-59)
+        "RR": "%y",  # 15
+        "YY": "%y",  # 15
+        "YYYY": "%Y",  # 2015
+        "FF6": "%f",  # only 6 digits are supported in python formats
+    }
+
+    class Parser(MySQL.Parser):
+        FUNCTIONS = {
+            **MySQL.Parser.FUNCTIONS,
+            "TO_DATE": build_formatted_time(exp.StrToDate, "singlestore"),
+            "TO_TIMESTAMP": build_formatted_time(exp.StrToTime, "singlestore"),
+            "TO_CHAR": build_formatted_time(exp.TimeToStr, "singlestore"),
+            "STR_TO_DATE": build_formatted_time(exp.StrToDate, "mysql"),
+            "DATE_FORMAT": build_formatted_time(exp.TimeToStr, "mysql"),
+            # The first argument is converted to TIME(6)
+            # This is needed because exp.TimeToStr is converted to DATE_FORMAT
+            # which interprets the first argument as DATETIME and fails to parse
+            # string literals like '12:05:47' without a date part.
+            "TIME_FORMAT": lambda args: exp.TimeToStr(
+                this=exp.Cast(
+                    this=seq_get(args, 0),
+                    to=exp.DataType.build(
+                        exp.DataType.Type.TIME,
+                        expressions=[exp.DataTypeParam(this=exp.Literal.number(6))],
+                    ),
+                ),
```

Comment:
Is this correct? From the docs:

> `TIME_FORMAT (timeobj, format)`
> `timeobj`: a valid time, date, datetime, or parsable date string

What if you get a datetime value (string or not) and then you cast it into a `TIME(6)`? Also, curious why the size 6 was used specifically.

Reply:
`What if you get a datetime value (string or not) and then you cast it into a TIME(6)?` 
In this case, the date part will be deleted.
`Also, curious why the size 6 was used specifically.`
6 is the maximum precision of time in SingleStore. It was used to preserve microseconds and don't lose information.
I tested it more and found several situations when it can return incorrect results. I'm going to change this.

Reply:
In my testing, I used format symbols for the date part, but according to the docs 
```
format: a string containing format symbols. Only format symbols for the hour, minute, second, and microsecond parts are valid.
```
So this solution seems to work well on the valid queries.
The date part (which is not needed for formatting) is deleted.

Reply:
Ok, so the argument is that the `CAST` should be fine since `TIME_FORMAT` truncates non-time related segments in the input as well? Can casting fail on any of the values in the domain of `TIME_FORMAT`?